src/mqt/bench/cli.py

```diff
@@ -23,56 +26,127 @@ def format_help(self) -> str:
+def parse_benchmark_name_and_instance(algorithm: str) -> tuple[str, str | None]:
+    """Parse an algorithm name like "shor_xlarge" into a benchmark and instance name."""
+    if algorithm.startswith("shor_"):
+        parts = algorithm.split("_", 1)
+        return parts[0], parts[1]
+    return algorithm, None
+
 
-    The resulting QASM string is printed to the standard output stream;
-    no other output is generated.
-    """
+def main() -> None:
+    """Generate a single benchmark and output in specified format."""
     parser = CustomArgumentParser(description="Generate a single benchmark")
     parser.add_argument(
         "--level",
         type=str,
-        help='Level to generate benchmarks for ("alg", "indep", "nativegates" or "mapped")',
+        choices=["alg", "indep", "nativegates", "mapped"],
+        help='Level to generate benchmarks for ("alg", "indep", "nativegates" or "mapped").',
+        required=True,
+    )
+    parser.add_argument(
+        "--algorithm",
+        type=str,
+        help="Name of the benchmark (e.g., 'grover-v-chain', 'shor_xsmall').",
         required=True,
     )
-    parser.add_argument("--algorithm", type=str, help="Name of the benchmark", required=True)
-    parser.add_argument("--num-qubits", type=int, help="Number of Qubits", required=True)
-    parser.add_argument("--qiskit-optimization-level", type=int, help="Qiskit compiler optimization level")
-    parser.add_argument("--native-gate-set", type=str, help="Name of the provider")
-    parser.add_argument("--device", type=str, help="Name of the device")
+    parser.add_argument(
+        "--num-qubits",
+        type=int,
+        help="Number of qubits for the benchmark.",
+        required=True,
+    )
+    parser.add_argument(
+        "--qiskit-optimization-level",
+        type=int,
+        help="Qiskit compiler optimization level (0-3).",
+    )
+    parser.add_argument(
+        "--native-gate-set",
+        type=str,
+        help="Provider name for native gate set (e.g., 'ibm', 'rigetti').",
+    )
+    parser.add_argument(
+        "--device",
+        type=str,
+        help="Device name for mapping stage (e.g., 'ibm_washington').",
+    )
+    parser.add_argument(
+        "--output-format",
+        type=str,
+        choices=[fmt.value for fmt in OutputFormat],
+        default=OutputFormat.QASM2.value,
+        help="Output format: 'qasm2', 'qasm3', or 'qpy'.",
+    )
+    parser.add_argument(
+        "--target-directory",
+        type=str,
+        default=".",
+        help="Directory to save the output file (only used for 'qpy' or if --save is specified).",
+    )
+    parser.add_argument(
+        "--target-filename",
+        type=str,
+        default="",
+        help="Base filename (without extension) for saved output.",
+    )
+    parser.add_argument(
+        "--save",
+        action="store_true",
+        help="If set, save the output to a file instead of printing to stdout (required for 'qpy').",
+    )
```

Comment:
I like this addition of writing to a file from the CLI instead of printing to stdout.
I'd probably slightly rephrase the existing help text for the `save` flag to just mention `qpy` as one of the examples that is not available as text-based output.
Additionally, I do not think that the `--target-filename` argument is necessary or helpful, but maybe I am overlooking things. MQT Bench has its naming scheme for files and I am not sure the user should be allowed to mess with it.
Of course, there needs to be some way for the user to specify where things should get stored.